View Controllers/ListViewController.m

```diff
@@ -21,69 +25,90 @@ @interface ListViewController ()
 @property (strong, nonatomic) NSArray<GMSPlace*>* favorites;
 @property (strong, nonatomic) NSArray<GMSPlace*>* wishlist;
 @property (assign, nonatomic) long segmentIndex;
+@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *progressIndicator;
+@property (strong, nonatomic) UIRefreshControl* refreshControl;
 
 @end
 
 @implementation ListViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-    // Do any additional setup after loading the view.
-    
-    [[NSNotificationCenter defaultCenter] addObserver:self
-                                             selector:@selector(setView)
-                                                 name:@"AddFavoriteNotification"
-                                               object:nil];
-    // add notification listener for adding to wishlist
-    [[NSNotificationCenter defaultCenter] addObserver:self
-                                             selector:@selector(setView)
-                                                 name:@"AddToWishlistNotification"
-                                               object:nil];
+    [self addNotificationObservers];
+    
+    // initialize arrays
+    self.favorites = [NSArray new];
+    self.wishlist = [NSArray new];
     
+    // set up tableview
     self.tableView.delegate = self;
     self.tableView.dataSource = self;
     self.tableView.hidden = NO;
```

Comment:
I'm seeing that in the code you've modified, and here, there are a lot of places that set the tableview hidden (or visible) and hide (or show) the default view. 

Is there a way we could clean up this code so we're not duplicating this flipping logic in many places? 

e.g. a helper method called something like "toggleDefaultView:(BOOL)shouldShowDefaultView)